src/frontend/serialization.rs

```diff
@@ -30,10 +30,9 @@ impl TryFrom<SignedPodHelper> for SignedPod {
             return Err(anyhow::anyhow!("pod_type is not Mock"));
```

Comment:
What about deserializing a non-mock SignedPod?
Would this implementation have a switch here?

Same for the MainPod.

Reply:
When I first wrote this, there were no non-mock SignedPods! Are the new non-mock Pods usable enough for me to implement serialization for them?

Reply:
Both the non-mock SignedPod and MainPod should be ready enough to get serialization and deserialization :)
I would suggest doing that in another PR though.